erts/emulator/beam/jit/x86/instr_common.cpp

```diff
@@ -1187,15 +1187,11 @@ void BeamModuleAssembler::emit_i_is_tuple(const ArgVal &Fail,
         /* Fast path for the `error | {ok, Value}` case. */
         comment("simplified tuple test since the source is always a tuple "
                 "when boxed");
-        a.test(getArgRef(Src, sizeof(byte)),
-               imm(_TAG_PRIMARY_MASK - TAG_PRIMARY_BOXED));
-
-        /* Note: ARG2 will NOT be set. This is OK since the operand
-         * for `current_tuple` has a type; that operand will not match
-         * the type-less operand for `get_tuple_element`. Thus, there
-         * will always be a `load_tuple_ptr` instruction emitted if
-         * this instruction is immediately followed by a
-         * `get_tuple_element` instruction. */
+        /* We must be careful to still leave the pointer to the tuple
+         * in ARG2. */
+        mov_arg(ARG2, Src);
+        (void)emit_ptr_val(ARG2, ARG2);
+        a.test(ARG2.r8(), imm(_TAG_PRIMARY_MASK - TAG_PRIMARY_BOXED));
```

Comment:
Swapping these improves latency a teeny tiny bit:
```suggestion
        a.test(ARG2.r8(), imm(_TAG_PRIMARY_MASK - TAG_PRIMARY_BOXED));
        (void)emit_ptr_val(ARG2, ARG2);
```

Reply:
But `emit_ptr_val()` will do an `and` when literals are tagged (always on Windows), which will disturb ZF.